tests/transforms/test_broadcast_expand.py

```diff
@@ -174,54 +151,52 @@ def test_not_copied(self):
-    def test_autograd(self, params, size, obs, exp_fn):
+    @pytest.mark.parametrize("diff_method", ["parameter-shift", "backprop"])
+    def test_autograd(self, params, obs, exp_fn, diff_method):
         """Test that the expansion works with autograd and is differentiable."""
         params = tuple(pnp.array(p, requires_grad=True) for p in params)
 
+        @qml.transforms.broadcast_expand
+        @qml.qnode(dev, interface="autograd", diff_method=diff_method)
         def cost(*params):
-            tape = make_tape(*params, obs)
-            tapes, fn = qml.transforms.broadcast_expand(tape)
-            if len(tape.measurements) > 1:
-                return qml.math.stack(fn(qml.execute(tapes, dev, qml.gradients.param_shift)))
-            return fn(qml.execute(tapes, dev, qml.gradients.param_shift))
+            make_ops(*params)
+            return qml.math.stack([qml.expval(ob) for ob in obs])
 
         expected = exp_fn(*params)
 
-        if len(obs) > 1 and size == 1:
-            expected = expected.T
-
         assert qml.math.allclose(cost(*params), expected)
 
         jac = qml.jacobian(cost)(*params)
         exp_jac = qml.jacobian(exp_fn)(*params)
 
-        if len(obs) > 1 and size == 1:
-            new_exp_jac = [qml.math.flatten(el) for el in exp_jac]
-            new_jac = [qml.math.flatten(el) for el in jac]
-            assert all(qml.math.allclose(_jac, e_jac) for _jac, e_jac in zip(new_jac, new_exp_jac))
-        else:
-            assert all(qml.math.allclose(_jac, e_jac) for _jac, e_jac in zip(jac, exp_jac))
+        assert all(qml.math.allclose(_jac, e_jac) for _jac, e_jac in zip(jac, exp_jac))
 
     @pytest.mark.jax
-    @pytest.mark.parametrize("params, size", parameters_and_size)
+    @pytest.mark.parametrize("params", parameters)
     @pytest.mark.parametrize("obs, exp_fn", observables_and_exp_fns)
-    def test_jax(self, params, size, obs, exp_fn):
+    @pytest.mark.parametrize("use_jit", [True, False])
+    @pytest.mark.parametrize("diff_method", ["parameter-shift", "backprop"])
+    def test_jax(self, params, obs, exp_fn, use_jit, diff_method):
         """Test that the expansion works with jax and is differentiable."""
+        # pylint: disable=too-many-arguments
         import jax
 
+        jax.config.update("jax_enable_x64", True)
+
         params = tuple(jax.numpy.array(p) for p in params)
 
+        @qml.transforms.broadcast_expand
+        @qml.qnode(dev, interface="jax", diff_method=diff_method)
         def cost(*params):
-            tape = make_tape(*params, obs)
-            tapes, fn = qml.transforms.broadcast_expand(tape)
-            return fn(qml.execute(tapes, dev, qml.gradients.param_shift))
+            make_ops(*params)
+            return tuple(qml.expval(ob) for ob in obs)
 
-        expected = exp_fn(*params)
+        if use_jit:
+            cost = jax.jit(cost)
 
-        if len(obs) > 1 and size == 1:
-            expected = expected.T
```

Comment:
The extraction of the first entry of `out` in the `exp_fn`s takes care of this already. There is nothing to transpose because in this case there only is one axis in `expected`.